user-interface/src/main/java/life/qbic/datamanager/views/projects/project/access/ProjectAccessComponent.java

```diff
@@ -154,28 +155,30 @@ private void loadInformationForProject(ProjectId projectId) {
     loadProjectAccessibleRoles(projectId);
   }
 
+  //shows active users in the UI
   private void loadProjectAccessibleUsers(ProjectId projectId) {
-    List<String> usernames = projectAccessService.listUsernames(projectId);
-    List<QbicUserDetails> users = usernames.stream()
-        .map(it -> (QbicUserDetails) userDetailsService.loadUserByUsername(it))
-        .distinct()
-        .toList();
-    List<String> authorities = projectAccessService.listAuthorities(projectId).stream()
-        .distinct()
-        .toList();
-    var entries = users.stream()
-        .map(userDetail -> {
-          var roles = getProjectRoles(authorities, userDetail);
-          roles = roles.stream()
-              .map(this::formatAuthorityToReadableString)
-              .toList();
-          String fullName = userInformationService.findById(userDetail.getUserId()).get().fullName();
-          return new UserProjectAccess(fullName, userDetail.getUsername(), String.join(", ", roles));
-        })
+    List<String> userIds = projectAccessService.listActiveUserIds(projectId);
+    List<QbicUserDetails> users = new ArrayList<>();
+    for(String id : userIds) {
+      Optional<UserInfo> optionalInfo = userInformationService.findById(id);
+      if(optionalInfo.isPresent()) {
+        QbicUserDetails userDetails = (QbicUserDetails) userDetailsService.
+            loadUserByUsername(optionalInfo.get().emailAddress());
+        users.add(userDetails);
+      }
+    }
+    List<String> authorities = projectAccessService.listAuthorities(projectId).stream().distinct()
         .toList();
+    var entries = users.stream().map(userDetail -> {
+      var roles = getProjectRoles(authorities, userDetail);
+      roles = roles.stream()
+          .map(this::formatAuthorityToReadableString)
+          .toList();
+      String fullName = userInformationService.findById(userDetail.getUserId()).get().fullName();
+      return new UserProjectAccess(fullName, userDetail.getEmailAddress(), String.join(", ", roles));
```

Comment:
Since this is our implementation, why do we need the email address here and further use the concept user name?